src/sentry/tasks/codeowners/code_owners_auto_sync.py

```diff
@@ -0,0 +1,52 @@
+from typing import TYPE_CHECKING
+
+from sentry.notifications.notifications.codeowners_auto_sync import AutoSyncNotification
+from sentry.tasks.base import instrumented_task
+
+if TYPE_CHECKING:
+    from sentry.models.commit import Commit
+
+
+@instrumented_task(
+    name="sentry.tasks.code_owners_auto_sync",
+    queue="code_owners",
+    # If the task fails, the user can manually sync. We'll send them an email on failure.
+    max_retries=0,
+)
+def code_owners_auto_sync(commit: Commit, **kwargs):
```

Comment:
@NisanthanNanthakumar So tasks should _not_ take an instance of a Django role as an input as it will use pickle serialization which we generally try to avoid. Instead, you can pass the ID of the commit as an argument and query for in the task.

Reply:
oh nice! TIL!